### PR TITLE
refactor(datepicker): replace SimpleDate & CalendarLocale with DateAdapter

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/lib/datepicker/year-view.spec.ts [hunk 1]
```diff
@@ -2,11 +2,16 @@ import {async, ComponentFixture, TestBed} from '@angular/core/testing';
...
 import {MdYearView} from './year-view';
-import {SimpleDate} from '../core/datetime/simple-date';
 import {MdCalendarBody} from './calendar-body';
 import {DatetimeModule} from '../core/datetime/index';
 
 
+// When constructing a Date, the month is zero-based. This can be confusing, since people are
+// used to seeing them one-based. So we create these aliases to make reading the tests easier.
+const JAN = 0, FEB = 1, MAR = 2, APR = 3, MAY = 4, JUN = 5, JUL = 6, AUG = 7, SEP = 8, OCT = 9,
+      NOV = 10, DEC = 11;
```
Comment (kara) [inline src/lib/datepicker/year-view.spec.ts]: This does help a ton 👍  

You have the mapping in a few different spec pages though - have you considered sharing it? Maybe under `Months.JAN` or something?

File: src/lib/datepicker/month-view.ts [hunk 2]
```diff
@@ -77,10 +76,14 @@ export class MdMonthView implements AfterContentInit {
   /** The names of the weekdays. */
   _weekdays: string[];
 
-  constructor(private _locale: CalendarLocale) {
+  constructor(public _dateAdapter: DateAdapter<D>) {
+    let firstDayOfWeek = this._dateAdapter.getFirstDayOfWeek();
+    let weekdays = this._dateAdapter.getDayOfWeekNames('narrow');
```
Comment (kara) [inline src/lib/datepicker/month-view.ts]: Nit: const?

File: src/lib/datepicker/year-view.spec.ts (hunk 1 above)
Comment (jelbourn) [inline src/lib/datepicker/year-view.spec.ts]: It's so much more readable to just redeclare the constants that it's worth the repetition.

File: src/lib/datepicker/datepicker-l10n.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+
+
+@Injectable()
+/** Datepicker data that requires localization. */
+export class MdDatepickerL10n {
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker-l10n.ts]: What do you think of `MdDatepickerIntl`?

File: src/lib/datepicker/datepicker-l10n.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+
+
+@Injectable()
+/** Datepicker data that requires localization. */
```
Comment (jelbourn) [inline src/lib/datepicker/datepicker-l10n.ts]: Comment should go above the decorator

**Pull request merged**